closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityProperties.java

```diff
@@ -0,0 +1,765 @@
+                "the restricted security mode, including the existing default properties and the " +
+                "restricted security properties.");
+        System.out.println("-Dsemeru.restrictedsecurity=help  This flag will print help message.");
+
+        System.out.println();
+        System.out.println("e.g.");
+        System.out.println("    -Dsemeru.restrictedsecurity=1,trace,audit,help");
+        System.out.println("    -Dsemeru.restrictedsecurity=help");
+
+        System.out.println();
+    }
+
+    /**
+     * Check if the input string is null and empty.
+     *
+     * @param string the input string
+     * @return true if the input string is null and emtpy
+     */
+    private static boolean isNullOrBlank(String string) {
+        return (string == null) || string.isBlank();
+    }
+
+    /**
+     * Check if the input string is null. If null return "".
+     *
+     * @param string the input string
+     * @return "" if the string is null
+     */
+    private static String parseProperty(String string) {
+        return string != null ? string.trim() : "";
+    }
+
+    /**
+     * Function to check if brackets are balanced.
+     *
+     * @param string Input string for checking
+     * @return true if the brackets are balanced
+     */
+    private boolean areBracketsBalanced(String string) {
+
+        Deque<Character> deque = new LinkedList<>();
+
+        for (char ch : string.toCharArray()) {
+            if (ch == '{' || ch == '[' || ch == '(') {
+                deque.addFirst(ch);
+            } else if (ch == '}' || ch == ']' || ch == ')') {
+                if (!deque.isEmpty()
+                        && ((deque.peekFirst() == '{' && ch == '}')
+                                || (deque.peekFirst() == '[' && ch == ']')
+                                || (deque.peekFirst() == '(' && ch == ')'))) {
+                    deque.removeFirst();
+                } else {
+                    return false;
+                }
+            }
+        }
+        return deque.isEmpty();
+    }
+
+    /**
+     * Nested class for provider's constraints
+     */
+    private static final class Constraint {
+        final String type;
+        final String algorithm;
+        final String attributes;
+
+        Constraint(String type, String algorithm, String attributes) {
+            super();
+            this.type = type;
+            this.algorithm = algorithm;
+            this.attributes = attributes;
+        }
+
+        String getType() {
+            return type;
+        }
+
+        String getAlgorithm() {
+            return algorithm;
+        }
+
+        String getAttributes() {
+            return attributes;
+        }
+    }
+
+    public String getDescName() {
+        return descName;
+    }
+
+    public String getDescNumber() {
+        return descNumber;
+    }
+
+    public String getDescPolicy() {
+        return descPolicy;
+    }
```

Comment:
Public functions need javadoc and these should have better names (at least expand "Desc" to "Description" or similar).